### PR TITLE
Add further unit tests for io.restassured.RestAssured

### PR DESCRIPTION
**Pull request by petersear**
Diffblue people: I've moved the previous old-style named tests for form() into a new test, testForm(), along with some new asserts. I've also added tests for other methods.

Hello. I've had another look at io.restassured.RestAssured and written some more unit tests with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes.

### CODE REVIEW

File: rest-assured/src/test/java/io/restassured/RestAssuredTest.java [hunk 1]
```diff
@@ -26,26 +28,51 @@
...
-    // Arrange
-    final String userName = "    ";
-    final String password = null;
-    final FormAuthConfig config = null;
-    // Act
-    thrown.expect(IllegalArgumentException.class);
-    RestAssured.form(userName, password, config);
-    // Method is not expected to return due to exception thrown
+  public void testBasic() {
+    Assert.assertNotNull(RestAssured.basic("user", "pw"));
```
Comment (peterschrammel) [inline rest-assured/src/test/java/io/restassured/RestAssuredTest.java:32]: The assertions seem very weak. Add deeper assertions on the return value.

File: rest-assured/src/test/java/io/restassured/RestAssuredTest.java [hunk 2]
```diff
@@ -26,26 +28,51 @@
   @Rule public ExpectedException thrown = ExpectedException.none();
 
   @Test
-  public void formInputNotNullNullNullOutputIllegalArgumentException() {
```
Comment (peterschrammel) [inline rest-assured/src/test/java/io/restassured/RestAssuredTest.java:29]: Please do the refactoring in a separate commit.

Comment (petersear) [reply rest-assured/src/test/java/io/restassured/RestAssuredTest.java:29]: Refactoring now in first commit.
New tests added in second commit.

File: rest-assured/src/test/java/io/restassured/RestAssuredTest.java (hunk 1 above)
Comment (peterschrammel) [reply rest-assured/src/test/java/io/restassured/RestAssuredTest.java:32]: That's quite difficult test, but you could try to do the following:
Call the `authenticate` method on the return value with a mocked `HttpBuilder` object. Then use `Mockito.verify` to check whether it is passed the right arguments.
